charts/rasa-x/templates/_helpers.tpl

```diff
@@ -210,3 +219,53 @@ Return 'true' if an enterprise version is run.
 {{- print "false" -}}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Return the rasa-x.version value as a default if the dbMigrationService.tag variable is not defined.
+*/}}
+{{- define "db-migration-service.version" -}}
+{{ .Values.dbMigrationService.tag | default (include "rasa-x.version" .) }}
+{{- end -}}
+
+{{/*
+Return 'true' if required version to run the database migration service is correct.
+If version is not valid semantic version then not use the DB migration service.
+*/}}
+{{- define "db-migration-service.requiredVersion" -}}
+{{- if and (not (regexMatch "^(v[0-9]|0|[1-9]\\d*).(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$" (include "db-migration-service.version" .))) (not .Values.dbMigrationService.ignoreVersionCheck) -}}
+{{- print "false" -}}
+{{- else if .Values.dbMigrationService.ignoreVersionCheck  -}}
+{{- print "true" -}}
+{{- else -}}
+{{- if semverCompare ">= 0.33.0" (include "db-migration-service.version" .) -}}
+{{- print "true" -}}
+{{- else -}}
+{{- print "false" -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return an init container for database migration.
+*/}}
+{{- define "initContainer.dbMigration" -}}
+{{ if and (eq "true" (include "db-migration-service.requiredVersion" .)) .Values.separateDBMigrationService }}
+initContainers:
+- name: init-db
+  image: {{ .Values.dbMigrationService.initContainer.image }}
+  command:
+  - 'sh'
+  - '-c'
+  - "apk update --no-cache && \
+    apk add jq curl && \
+    until nslookup {{ .Release.Name }}-db-migration-service-headless 1> /dev/null; do echo Waiting for the database migration service; sleep 2; done && \
+    until [[ \"$(curl -s http://{{ .Release.Name }}-db-migration-service-headless:{{ .Values.dbMigrationService.port }} | jq -r .status)\" == \"completed\" ]]; do \
```

Comment:
should we use a template for `http://{{ .Release.Name }}-db-migration-service-headless:{{ .Values.dbMigrationService.port }}` as we are repeating this multiple times?